src/hal/lib/hal_comp.c

```diff
@@ -372,36 +372,39 @@ const char *hal_comp_name(int comp_id)
 
 // use only for owner_ids of pins, params or functs
 // may return NULL if buggy using code
+// This was a bottleneck with rcomps validating pins
+// Move checks so that instances get checked first
+
 hal_comp_t *halpr_find_owning_comp(const int owner_id)
 {
-    hal_comp_t *comp = halpr_find_comp_by_id(owner_id);
-    if (comp != NULL)
-	return comp;  // legacy case: the owner_id refers to a comp
+hal_comp_t *comp;
 
-    // nope, so it better be an instance
+    // more likely to be instcomp, so test that first
     hal_inst_t *inst = halpr_find_inst_by_id(owner_id);
-    if (inst == NULL) {
-	HALERR("BUG: owner_id %d refers neither to a hal_comp_t nor an hal_inst_t",
-	       owner_id);
-	return NULL;
-    }
 
+    if (inst == NULL) { // is it a legacy comp?
+        comp = halpr_find_comp_by_id(owner_id);
+        if (comp != NULL)
```

Comment:
You seem to have mistaken my merging of your odyssey of autistic 'deckchair rearrangement' at https://github.com/machinekit/machinekit/commit/b0f2539128931a8f32c1046b0be053913a9b39ac as approval for it or validation.

It should be regarded as an indication of my tolerance.
That I am unafflicted by a certainty of the  'one true method of code styling', complete with links to show how 'right' I am (even if one of them is talking about something different).
Plus a recognition that if I tried to impose such views on others , I would be pretty soon be left to get on with all the work on my own.

Something you should think upon.